server/src/main/java/org/elasticsearch/index/translog/TranslogWriter.java

```diff
@@ -399,8 +399,12 @@ final boolean syncUpTo(long offset) throws IOException {
                             ensureOpen();
                             checkpointToSync = getCheckpoint();
                             toWrite = pollOpsToWrite();
-                            flushedSequenceNumbers = nonFsyncedSequenceNumbers;
-                            nonFsyncedSequenceNumbers = new LongArrayList(64);
+                            if (nonFsyncedSequenceNumbers.isEmpty()) {
+                                flushedSequenceNumbers = null;
```

Comment:
Is there an existing test provoking this case or can we add one?

Reply:
Same here, lots of empty syncing in recovery etc. here that triggers this.